enterprise_catalog/apps/api/v1/views/enterprise_customer.py

```diff
@@ -125,31 +135,11 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
         if (get_catalogs_containing_specified_content_ids or get_catalog_list):
             response_data['catalog_list'] = catalogs_that_contain_course
 
-        response_data[RESTRICTED_RUNS_ALLOWED_KEY] = self._get_restricted_runs_allowed_for_query(
-            course_run_ids, customer_catalogs,
+        response_data[RESTRICTED_RUNS_ALLOWED_KEY] = get_restricted_runs_allowed_for_query(
+            course_run_ids, customer_catalogs
```

Comment:
I think after some more rubber ducking today, I think we might want to pass `catalogs_that_contain_course` to `get_restricted_runs_allowed_for_query()` instead of all of the customer's catalogs. This avoids needing to cover an edge case in the frontend where the parent course does not match the base query, but one of its restricted runs does. We would use help texts and documentation to guide ECS to only write custom filters that match both the course and any of its restricted runs.
```suggestion
        response_data[RESTRICTED_RUNS_ALLOWED_KEY] = get_restricted_runs_allowed_for_query(
            course_run_ids, catalogs_that_contain_course
```